File: business-services/client-api/src/test/java/com/ocean/bank/clien/controller/ClientControllerTest.java
//package com.ocean.bank.clien.controller;
//
//import com.ocean.bank.client.controller.dto.ClientInfo;
//import com.ocean.bank.client.repository.entity.Address;
//import com.ocean.bank.client.repository.entity.IdentificationDocument;
//import com.ocean.bank.client.repository.entity.SensitiveData;
//import org.junit.jupiter.api.Test;
//
//import static org.assertj.core.api.Assertions.assertThat;
//import static org.junit.jupiter.api.Assertions.assertAll;
//
//class ClientControllerTest extends BaseControllerTest {
//
//    @Test
//    void createClient() {
//        var clientInfo = ClientInfo.builder()
//                .firstName("John")
//                .lastName("Doe")
//                .identificationDocument(
//                        IdentificationDocument.builder().documentName("passport").number("123456").code("123456").build())
//                .address(Address.builder().registrationAddress("registrationAddress").build())
//                .sensitiveData(SensitiveData.builder().passphrase("passphrase").password("password").build())
//                .phoneNumber("123456")
//                .build();
//
//
//        var response = webTestClient.post()
//                .uri("")
//                .bodyValue(clientInfo)
//                .exchange()
//                .expectStatus().isOk();
//
//        assertAll(
//                () -> assertThat(response.expectBody().jsonPath("firstName").isEqualTo("John")),
//                () -> assertThat(response.expectBody().jsonPath("lastName").isEqualTo("Doe")),
//                () -> assertThat(response.expectBody().jsonPath("clientCode").isNotEmpty()),
//                () -> assertThat(response.expectBody().jsonPath("status").isNotEmpty())
//        );
//    }
//}

File: account-api/src/test/kotlin/com/ocean/bank/account/controller/ClientControllerTest.kt
package com.ocean.bank.account.controller

import com.ocean.bank.account.repository.JDBCTemplateClientDao
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertAll
import org.springframework.beans.factory.annotation.Autowired

class ClientControllerTest : BaseIntegrationTest() {

    @Autowired
    private lateinit var clientRepository: JDBCTemplateClientDao

    @Test
    fun addClientTest() {
        webTestClient.post()
            .uri { it.path(CLIENT_PATH).build() }
            .header("client-code", CLIENT_CODE_2)
            .exchange()
            .expectStatus().isOk

        // FIXME: а тут почему-то могу получить клиента напрямую из БД, в отличии от счетов
        val client = clientRepository.findClientByCode(CLIENT_CODE_2)
        assertAll(
            { assertThat(client).isNotNull },
            { assertThat(client?.code).isEqualTo("AAAAB") },
            { assertThat(client?.statusId).isEqualTo(1) }
        )
    }

    @Test
    fun updateClientStatusTest() {
        clientRepository.saveClient(createClient())

        webTestClient.put()
            .uri { it.path(CLIENT_PATH).queryParam("status", "ACTIVE").build() }
            .header("client-code", CLIENT_CODE)
            .exchange()
            .expectStatus().isOk

        val client = clientRepository.findClientByCode(CLIENT_CODE)
        assertAll(
            { assertThat(client).isNotNull },
            { assertThat(client?.code).isEqualTo(CLIENT_CODE) },
            { assertThat(client?.statusId).isEqualTo(2) }
        )
    }

    companion object {
        const val CLIENT_PATH = "internal/client"
    }
}